src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
 package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/teams"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Channels() Channels {
+	return Channels{c}
+}
+
+// Channels is an interface-compliant provider of the client.
+type Channels struct {
+	Client
+}
+
+// ---------------------------------------------------------------------------
+// containers
+// ---------------------------------------------------------------------------
+
+// CreateContainer makes an channels with the name in the team
+func (c Channels) CreateChannel(
+	ctx context.Context,
+	teamID, _, containerName string,
+) (graph.Container, error) {
+	body := models.NewChannel()
+	body.SetDisplayName(&containerName)
+
+	container, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating channel")
+	}
+
+	return ChannelsDisplayable{Channelable: container}, nil
+}
+
+// DeleteChannel removes a channel from user's M365 account
+func (c Channels) DeleteChannel(
+	ctx context.Context,
+	teamID, containerID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	return nil
+}
+
+// prefer GetChannelByID where possible.
```

Comment:
In the other cases where we've commented `prefer GetFooByID`, Foo is either Item or Container.  The comment effectively means "use the generic interface if you can".  (for that matter, we could probably get rid of one func by use of generics).  But since `GetChannelByID` is named for `Channel` and not Item or Container, there is no generic interface compliance, and so this function is unnecessary duplication.